### PR TITLE
Travis build only needed

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Targeting and hopefully closing #625

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Hm. So I'm not sure whether this actually worked the way we intended. I guess I will do the following: Squash this branch, make the history nice. Then I will squash it again into one TMP commit and randomly apply this commit to other branches to see whether they build correctly. (and of course revert these patches later on again).

**Comment by matthiasbeyer** [comment]
@TheNeikos and @neithernut and maybe even @manuelmessner - could you have a close look? I hope this is correct... and yes, it is my bad bash stlye, but it ~~works~~ runs (as reported by travis)! :smile:

File: .travis_build_script_functions.sh [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    dbg "Getting metadata for $1, grepping $2"
+    cargo metadata --manifest-path "$1" | grep "\"$2\"" 2>/dev/null
+}
+
+# Gets all pathes to Cargo.toml files in changed directories in $*
+get_cargo_tomls_for_changed_dirs() {
+    dbg "Finding Cargo.toml in $*"
+    for d in $*; do
+        find "$d" -name "Cargo.toml" -type f
+    done
```
Comment (neithernut) [inline .travis_build_script_functions.sh]: You can pass `$*` to find directly

File: .travis_build_script.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+source ./.travis_build_script_functions.sh
+
+for changed_dir in $(get_changed_dirs); do
+    toml=$(get_cargo_tomls_for_changed_dirs $changed_dir);
+    tomls=$(reverse_dep_recursive $toml)
+    echo "$tomls"
+done
```
Comment (neithernut) [inline .travis_build_script.sh]: This loops does anything else but produce output?

Comment (matthiasbeyer) [inline .travis_build_script.sh]: Ah, nope. I will remove it.

**Comment by neithernut** [comment]
Looks like you have dead code in `.travis_build_script.sh`.

**Comment by matthiasbeyer** [comment]
> Looks like you have dead code in .travis_build_script.sh.

Why don't you annotate the dead code (I fail to see) - so I know where it is? :smile:

**Comment by neithernut** [comment]
Mh... you are right. They are pulled in by `get_to_build_tomls()`.

**Comment by matthiasbeyer** [comment]
It seems like I fu**** up here. I will close this unmerged - we already have a shorter build time,... should be enough for now.

**Pull request closed without merging**